pkg/datarepair/queue/queue.go

```diff
@@ -0,0 +1,82 @@
+// Copyright (C) 2018 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package queue
+
+import (
+	"encoding/binary"
+	"sync"
+	"time"
+
+	"github.com/golang/protobuf/proto"
+	"github.com/zeebo/errs"
+
+	"storj.io/storj/pkg/pb"
+	"storj.io/storj/storage"
+	"storj.io/storj/storage/redis"
+)
+
+//RepairQueue is the interface for the data repair queue
+type RepairQueue interface {
+	Add(qi *pb.InjuredSegment) (storage.Key, error)
+	Remove(date storage.Key) error
+	GetNext() (storage.Key, pb.InjuredSegment, error)
```

Comment:
`Remove`, `Add`, `GetNext` returning the storage.Key seems weird and combination seems weird. Why should the  enqueuer care about where and how it's stored. Similarly the user shouldn't care. It would be sufficient to have:

```
Add(qi *pb.InjuredSegment) error
GetNext() (pb.InjuredSegment, error)
```

Reply:
i added storage.Key here so that after the repairer got the next item off the queue, they'd be able to remove it from the queue. Is there a better way to do this?

Reply:
`GetNext` should automatically remove it, otherwise multiple repairers would get the same item (imagine 100 concurrent GetNext, which all get the same value... and then all try to concurrently remove it). You could rename the method pair to `Enqueue`/`Dequeue`; `Push`/`Pop` or `Send`/`Recv` to make it more obvious that getting the item from the queue is removing them.